src/nvim/os/win_defs.h

```diff
@@ -1,25 +1,88 @@
 #ifndef NVIM_OS_WIN_DEFS_H
 #define NVIM_OS_WIN_DEFS_H
 
+#include "auto/config.h"
+
+#ifndef LC_MESSAGES
+// FIXME: this is a hack - also used in ex_cmds2.c
+// to build in windows without libintl
+# define LC_MESSAGES LC_COLLATE
+#endif
+
+// winsock2.h must be before windows.h - or so says Mingw
+#include <winsock2.h>
 #include <windows.h>
+// We have our own RGB macro in macros.h
+#undef RGB
+
+#include <io.h>
+#include <uv.h>
+
+#include <stdio.h>
+#include <time.h>
+#include <sys/stat.h>
+
+// FIXME: this is only used by if_cscope.c
+#ifndef S_ISLNK
+# define S_ISLNK(mode) 0
+#endif
+
+// For MSVC
+#ifdef _MSC_VER
+# ifndef restrict
+#  define restrict __restrict
+# endif
+# ifndef inline
+#  define inline __inline
+# endif
+# ifndef __func__
+#  define __func__ __FUNCTION__
+# endif
```

Comment:
Hmm me too.

The problem is that for some reason `__func__` is defined when compiling in `C++` (compiler flag is `/TP`) mode but not in `C` (compiler flag is `/TC`) mode on MSVC.

MSDN seems to indicate it should work since it's listed as an C99 predefined macro.

See [here](https://msdn.microsoft.com/en-us/library/b0084kay.aspx).

Perhaps there is an open connect bug for this or we can open one and see what happens.

In the meantime I suppose we will have to stick with our own definition.